metadata-service/auth-impl/src/main/java/com/datahub/authentication/token/StatefulTokenService.java

```diff
@@ -0,0 +1,184 @@
+import com.linkedin.metadata.resources.entity.AspectUtils;
+import com.linkedin.metadata.utils.AuditStampUtils;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.MetadataChangeProposal;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.NotImplementedException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static com.datahub.authentication.token.TokenClaims.ACTOR_ID_CLAIM_NAME;
+import static com.datahub.authentication.token.TokenClaims.ACTOR_TYPE_CLAIM_NAME;
+import static com.datahub.authentication.token.TokenClaims.TOKEN_TYPE_CLAIM_NAME;
+import static com.datahub.authentication.token.TokenClaims.TOKEN_VERSION_CLAIM_NAME;
+
+
+/**
+ * Service responsible for generating JWT tokens for use within DataHub that are stored in the entity service
+ * so that we can list & revoke tokens as needed.
+ */
+@Slf4j
+public class StatefulTokenService extends StatelessTokenService {
+
+    private final EntityService _entityService;
+    private final LoadingCache<String, Boolean> _revokedTokenCache;
+    private final String saltingKey;
+    private final MessageDigest sha256;
+
+    public StatefulTokenService(@Nonnull final String signingKey, @Nonnull final String signingAlgorithm,
+                                @Nullable final String iss, @Nonnull final EntityService entityService, @Nonnull final String saltingKey) {
+        super(signingKey, signingAlgorithm, iss);
+        this._entityService = entityService;
+        this._revokedTokenCache = CacheBuilder.newBuilder()
+                .maximumSize(10000)
+                .expireAfterWrite(6, TimeUnit.HOURS)
+                .build(new CacheLoader<String, Boolean>() {
+                    @Override
+                    public Boolean load(final String key) {
+                        final Urn accessUrn = Urn.createFromTuple(Constants.ACCESS_TOKEN_ENTITY_NAME, key);
+                        return !_entityService.exists(accessUrn);
+                    }
+                });
+        this.saltingKey = saltingKey;
+        try {
+            this.sha256 = MessageDigest.getInstance("SHA-256");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Unable to get SHA-256 algorithm.");
+        }
+    }
+
+    /**
+     * Generates a JWT for an actor with a default expiration time.
+     *
+     * Note that the caller of this method is expected to authorize the action of generating a token.
+     *
+     */
+    public String generateAccessToken(@Nonnull final TokenType type, @Nonnull final Actor actor) {
```

Comment:
Is this an override? 

If so, then you should probably include the @Override annotation